src/clj_kondo/impl/analyzer.clj

```diff
@@ -523,16 +523,26 @@
                              %)
                            bodies)
         ;; poor naming, this is for type information
-        arities (into {} (map (fn [{:keys [:fixed-arity :varargs? :min-arity :ret :args]}]
-                                (let [arg-tags (when (some identity args)
-                                                 args)
-                                      v (assoc-some {}
-                                                    :ret ret :min-arity min-arity
-                                                    :args arg-tags)]
-                                  (if varargs?
-                                    [:varargs v]
-                                    [fixed-arity v]))))
-                      parsed-bodies)
+        arities (reduce (fn [acc {:keys [:fixed-arity :varargs? :min-arity :ret :args] :as arity}]
```

Comment:
we could extract this part into a single function used by both `analyze-fn` and `analyze-defn`. Not sure what the reasoning is behind the current duplication so i didn't do anything yet,  but happy to extract if desired.

Reply:
feel free to extract